File: android/app/src/main/java/com/beetle/bauhinia/Token.java
package com.beetle.bauhinia;

import com.google.code.p.leveldb.LevelDB;
import com.google.gson.annotations.SerializedName;

/**
 * Created by houxh on 14-8-11.
 */
public class Token {
    private static Token instance;
    public static Token getInstance() {
        if (instance == null) {
            instance = new Token();
            instance.load();
        }
        return instance;
    }

    @SerializedName("access_token")
    public String accessToken;
    @SerializedName("refresh_token")
    public String refreshToken;
    @SerializedName("expires_in")
    public int expireTimestamp;

    public void save() {
        LevelDB db = LevelDB.getDefaultDB();
        try {
            db.set("access_token", accessToken);
            db.set("refresh_token", refreshToken);
            db.setLong("token_expire", expireTimestamp);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private void load() {
        LevelDB db = LevelDB.getDefaultDB();
        try {
            accessToken = db.get("access_token");
            refreshToken = db.get("refresh_token");
            expireTimestamp = (int)db.getLong("token_expire");
        } catch(Exception e) {
            e.printStackTrace();
        }
    }
}

File: ios/Message/model/Token.m
//
//  Token.m
//  Message
//
//  Created by houxh on 14-7-8.
//  Copyright (c) 2014年 daozhu. All rights reserved.
//

#import "Token.h"
#import "LevelDB.h"
#import "TAHttpOperation.h"
#import "Config.h"
#import "APIRequest.h"

@interface Token()
@property(nonatomic)dispatch_source_t refreshTimer;
@property(nonatomic)int refreshFailCount;
@end

@implementation Token

+(Token*)instance {
    static Token *tok;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        if (!tok) {
            tok = [[Token alloc] init];
            [tok load];
        }
    });
    return tok;
}

-(id)init {
    self = [super init];
    if (self) {

    }
    return self;
}

-(void)load {
    LevelDB *db = [LevelDB defaultLevelDB];
    self.accessToken = [db stringForKey:@"access_token"];
    self.refreshToken = [db stringForKey:@"refresh_token"];
    self.expireTimestamp = (int)[db intForKey:@"token_expire"];
    self.uid = [db intForKey:@"token_uid"];
}

-(void)save {
    LevelDB *db = [LevelDB defaultLevelDB];
    [db setString:self.accessToken forKey:@"access_token"];
    [db setString:self.refreshToken forKey:@"refresh_token"];
    [db setInt:self.expireTimestamp forKey:@"token_expire"];
    [db setInt:self.uid forKey:@"token_uid"];
}

@end


@implementation TokenManager
RCT_EXPORT_MODULE();

- (dispatch_queue_t)methodQueue
{
    return dispatch_get_main_queue();
}

RCT_REMAP_METHOD(getToken,
                 resolver:(RCTPromiseResolveBlock)resolve
                 rejecter:(RCTPromiseRejectBlock)reject)
{
    
    if ([Token instance].uid > 0) {
        NSDictionary *dict =@{@"uid":@([Token instance].uid),
                              @"name":@"我",
                              @"gobelieveToken":[Token instance].accessToken,
                              };
        
        resolve(dict);
    } else {
        reject(@"token uninitialize", @"", nil);
    }
}


@end
